ui.lua

```diff
@@ -60,7 +60,26 @@ function ui.printcmd(rawline, ts, urgent_buf)
 				userpart = string.format("<%s>", hi(from))
 			end
 		end
-		print(prefix .. userpart .. " " .. msg)
+		if not config.left_margin then
+			print(prefix .. userpart .. " " .. msg)
+		else -- this is responsible for printing messages if the left_margin is enabled
+			-- yeah it's spaghetti, 
+			-- also i have no idea what 'notice' is so am copying that
+			-- t. juspib
+			if notice then
+				userpart = string.format("-%s:%s-", hi(from), to)
+			elseif action then
+				local vislen = string.len(os.date(config.timefmt, ts))
```

Comment:
I think you meant `string.len(prefix)` there. It might include more than the time, e.g. the buffer name if you got pinged in another channel.

Reply:
That being said, the margin probably should be disabled for messages with a buffer prefix?
Alternatively, maybe it could only show the buffer name, and the actual nick could be pushed to the right side?

Reply:
prefix can include ANSI escape sequences, which makes string.len(prefix) not relevant to how many columns the prefix occupies on a screen. Something like string.len(remove_ansi_escape_sequences(prefix)) would be optimal here.